s3plugin/s3plugin.go

```diff
@@ -99,6 +100,20 @@ func readAndValidatePluginConfig(configFile string) (*PluginConfig, error) {
 	return config, nil
 }
 
+func SetLoggerVerbosity(loggerVerbosity string) {
+	gplog.SetLogFileVerbosity(gplog.LOGINFO)
+	if loggerVerbosity == "error" {
```

Comment:
What about using a lookup table for possible verbosity cases? In this case the code could be more compact:

```diff
diff --git a/s3plugin/s3plugin.go b/s3plugin/s3plugin.go
index d058124..ccac3ec 100644
--- a/s3plugin/s3plugin.go
+++ b/s3plugin/s3plugin.go
@@ -73,6 +73,13 @@ type PluginOptions struct {
 	DownloadConcurrency int
 }
 
+var loggerVerbosityLut = map[string]int {
+	"info": gplog.LOGINFO,
+	"error": gplog.LOGERROR,
+	"debug": gplog.LOGDEBUG,
+	"verbose": gplog.LOGVERBOSE,
+}
+
 func CleanupPlugin(c *cli.Context) error {
 	return nil
 }
@@ -102,15 +109,11 @@ func readAndValidatePluginConfig(configFile string) (*PluginConfig, error) {
 
 func SetLoggerVerbosity(loggerVerbosity string) {
 	gplog.SetLogFileVerbosity(gplog.LOGINFO)
-	if loggerVerbosity == "error" {
-		gplog.SetVerbosity(gplog.LOGERROR)
-		gplog.SetLogFileVerbosity(gplog.LOGERROR)
-	} else if loggerVerbosity == "debug" {
-		gplog.SetVerbosity(gplog.LOGDEBUG)
-		gplog.SetLogFileVerbosity(gplog.LOGDEBUG)
-	} else if loggerVerbosity == "verbose" {
-		gplog.SetVerbosity(gplog.LOGVERBOSE)
-		gplog.SetLogFileVerbosity(gplog.LOGVERBOSE)
+
+	verbosityLevel, ok := loggerVerbosityLut[loggerVerbosity]
+	if ok {
+		gplog.SetVerbosity(verbosityLevel)
+		gplog.SetLogFileVerbosity(verbosityLevel)
 	}
 }
 
@@ -157,7 +160,8 @@ func InitializeAndValidateConfig(config *PluginConfig) error {
 	if opt.Encryption != "on" && opt.Encryption != "off" {
 		errTxt += fmt.Sprintf("Invalid encryption configuration. Valid choices are on or off.\n")
 	}
-	if opt.LoggerVerbosity != "error" && opt.LoggerVerbosity != "info" && opt.LoggerVerbosity != "verbose" && opt.LoggerVerbosity != "debug" {
+	_, loggerVerbosityOk := loggerVerbosityLut[opt.LoggerVerbosity]
+	if !loggerVerbosityOk {
 		errTxt += fmt.Sprintf("Invalid logger_verbosity configuration. Valid choices are error, info, verbose or debug.\n")
 	}
 	SetLoggerVerbosity(opt.LoggerVerbosity)
```

Reply:
I copied this function entirely from the [gpbackup](https://github.com/arenadata/gpbackup/blob/ae780919dc1e81fc0e085fc6372d19b66e1725fb/backup/wrappers.go#L32-L44).

Reply:
so any reason not to change it?

Reply:
applied